### ISSUE TITLE
Production builds crash. Android only (WindowManager).

### ISSUE THREAD

**Issue by NichAga** [rebeccahughes/react-native-device-info#342]
## Summary

| Version     | 0.17.2   |
| Affected OS | Android   |
| OS Version  | 5.0.2   |

## Current behavior
Debug builds work as expected, however release builds crash.
Here is the log:

> Java exception in 'NativeModules'
E/ReactNativeJS( 6156): 
E/ReactNativeJS( 6156): java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.WindowManager android.app.Activity.getWindowManager()' on a null object reference
D/ConnectivityManager.CallbackHandler( 6156): CM callback handler got msg 524290
D/ConnectivityService(  690): listenForNetwork for NetworkRequest [ id=32, legacyType=-1, [ Capabilities: INTERNET&NOT_RESTRICTED&TRUSTED] ]
D/ConnectivityService(  690): handleRegisterNetworkRequest checking NetworkAgentInfo [WIFI () - 100]
D/ConnectivityService(  690): apparently satisfied.  currentScore=57
E/AndroidRuntime( 6156): FATAL EXCEPTION: mqt_native_modules
E/AndroidRuntime( 6156): com.facebook.react.common.JavascriptException: Java exception in 'NativeModules'
E/AndroidRuntime( 6156): 
E/AndroidRuntime( 6156): java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.WindowManager android.app.Activity.getWindowManager()' on a null object reference, stack:
E/AndroidRuntime( 6156): com.learnium.RNDeviceInfo.RNDeviceModule.isTablet@99
E/AndroidRuntime( 6156): com.learnium.RNDeviceInfo.RNDeviceModule.getConstants@235
E/AndroidRuntime( 6156): com.facebook.react.bridge.JavaModuleWrapper.getConstants@140
E/AndroidRuntime( 6156): android.os.Handler.handleCallback@739
E/AndroidRuntime( 6156): android.os.Handler.dispatchMessage@95
E/AndroidRuntime( 6156): com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage@31
E/AndroidRuntime( 6156): <unknown>@426:52
E/AndroidRuntime( 6156): d@2:768
E/AndroidRuntime( 6156): n@2:409
E/AndroidRuntime( 6156): t@2:262
E/AndroidRuntime( 6156): <unknown>@397:343
E/AndroidRuntime( 6156): d@2:768
E/AndroidRuntime( 6156): n@2:409
E/AndroidRuntime( 6156): t@2:262
E/AndroidRuntime( 6156): <unknown>@396:2581
E/AndroidRuntime( 6156): d@2:768
E/AndroidRuntime( 6156): n@2:409
E/AndroidRuntime( 6156): t@2:262
E/AndroidRuntime( 6156): <unknown>@12:112
E/AndroidRuntime( 6156): d@2:768
E/AndroidRuntime( 6156): n@2:339
E/AndroidRuntime( 6156): t@2:262
E/AndroidRuntime( 6156): global code@775:8
E/AndroidRuntime( 6156): 
E/AndroidRuntime( 6156): 	at com.facebook.react.modules.core.ExceptionsManagerModule.showOrThrowError(ExceptionsManagerModule.java:56)
E/AndroidRuntime( 6156): 	at com.facebook.react.modules.core.ExceptionsManagerModule.reportFatalException(ExceptionsManagerModule.java:40)
E/AndroidRuntime( 6156): 	at java.lang.reflect.Method.invoke(Native Method)
E/AndroidRuntime( 6156): 	at java.lang.reflect.Method.invoke(Method.java:372)
E/AndroidRuntime( 6156): 	at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:374)
E/AndroidRuntime( 6156): 	at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:162)
E/AndroidRuntime( 6156): 	at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
E/AndroidRuntime( 6156): 	at android.os.Handler.handleCallback(Handler.java:739)
E/AndroidRuntime( 6156): 	at android.os.Handler.dispatchMessage(Handler.java:95)
E/AndroidRuntime( 6156): 	at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:31)
E/AndroidRuntime( 6156): 	at android.os.Looper.loop(Looper.java:135)
E/AndroidRuntime( 6156): 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$3.run(MessageQueueThreadImpl.java:194)
E/AndroidRuntime( 6156): 	at java.lang.Thread.run(Thread.java:818)

I don't know if it can help, but the only API I'm calling is getDeviceLocale().

I've reverted to the version 0.16.0 and it doesn't crash anymore. I don't know with 0.17 and 0.17.1

**Comment by machour** [comment]
Hi and thank you for reporting this!

```
E/AndroidRuntime( 6156): java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.WindowManager android.app.Activity.getWindowManager()' on a null object reference, stack:
E/AndroidRuntime( 6156): com.learnium.RNDeviceInfo.RNDeviceModule.isTablet@99
```

The problem seems to have been introduced in 0.17.2 with the modification to `isTablet()`.
Would you mind changing locally `getCurrentActivity()` to `getReactApplicationContext()` in this line
 https://github.com/rebeccahughes/react-native-device-info/blob/master/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java#L99
and tell me if it fixes the problem?

**Comment by NichAga** [comment]
error: cannot find symbol
    getReactApplicationContext().getWindowManager().getDefaultDisplay().getMetrics(metrics);

symbol:   method getWindowManager()

**Comment by machour** [comment]
@mikpalm do you have the time to look into this today by any chance?

**Comment by mikpalmen** [comment]
Ill have a look. right now.

**Comment by mikpalmen** [comment]
What react-native version are you running? just tested this with 0.48.4 and it works for me, but just so i know what to look for/at. @NichAga

**Comment by mikpalmen** [comment]
Updated the repo. @machour @NichAga 
changes made
line 98-99.
```
DisplayMetrics metrics = new DisplayMetrics();
getCurrentActivity().getWindowManager().getDefaultDisplay().getMetrics(metrics);
```
to
```
final Context ctx = getReactApplicationContext();
final DisplayMetrics metrics = ctx.getResources().getDisplayMetrics();
```

**Comment by machour** [comment]
@NichAga could you possibly test @mikpalmen suggestion and report back?

**Comment by NichAga** [comment]
@machour it works.

**Comment by machour** [comment]
Fixed in 0.17.3, many thanks to you both!

**Comment by NichAga** [comment]
@machour Sorry for the late reply, I was working on another thing but yes, it works.
